lib/takeaway.rb

```diff
@@ -0,0 +1,71 @@
+require 'twilio-ruby'
+
+class Takeaway
+  def initialize(menu)
+    @menu = menu
+    @basket = {}
+    @cost = OrderCost.new
+  end
+
+  attr_reader :basket
+
+  def total
+    "Total: £#{cost.total.to_f}0"
+  end
+
+  def check_total
+    cost.total == basket_sum
+  end
+
+  def menu
+    @menu.dishes
+  end
+
+  def order(dish, number=1)
+    @basket[ordered(dish,number)] = price(dish,number)
+    cost.add(price(dish,number))
+    "You have just ordered #{number}x #{dish}!"
+  end
+
+  def complete_order
+    fail "This order was not completed" if !check_total
```

Comment:
Style/NegatedIf: Favor `unless` over `if` for negative conditions. (https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)